packages/react/src/primitives/utils/useStableId.ts

```diff
@@ -0,0 +1,31 @@
+// Adapted from https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx#L8
+import * as React from 'react';
+import { useLayoutEffect } from './useLayoutEffect';
+
+// Prefixed autogenerated id created by useStableId
+export const AUTO_GENERATED_ID_PREFIX = 'amplify-id';
+
+export type UseStableId = (id?: string) => string;
```

Comment:
do we need to export it? there's no other references to `UseStableId` type on this PR

Reply:
It's used as the type of the `useStableId` function. My understanding is that if you export a function that uses a type, you must also export the type. Is that not the case?

Reply:
is not mandatory, but if you feel it's safer to export it, I'm good with that :)